stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/internal/MessageListItemDecoratorProvider.kt

```diff
@@ -22,18 +22,20 @@ internal class MessageListItemDecoratorProvider(
     style: MessageListItemStyle,
 ) : DecoratorProvider {
 
-    private val messageListDecorators = listOf<Decorator>(
+    private val messageListDecorators = mutableListOf<Decorator>(
         BackgroundDecorator(style),
         TextDecorator(style),
         GapDecorator(),
         MaxPossibleWidthDecorator(),
         AvatarDecorator(),
         FailedIndicatorDecorator(),
-        ReactionsDecorator(),
         ReplyDecorator(currentUser),
-        ReactionsDecorator(),
         FootnoteDecorator(dateFormatter, isDirectMessage),
-    )
+    ).apply {
+        if (style.reactionsEnabled) {
+            add(ReactionsDecorator())
+        }
+    }
```

Comment:
What about to not convert list to a mutable list?

You could do something like:
```
private val messageListDecorators = listOfNotNull<Decorator>(
        BackgroundDecorator(style),
        TextDecorator(style),
        GapDecorator(),
        MaxPossibleWidthDecorator(),
        AvatarDecorator(),
        FailedIndicatorDecorator(),
        ReactionsDecorator().takeIf { style.reactionsEnabled },
        ReplyDecorator(currentUser),
        FootnoteDecorator(dateFormatter, isDirectMessage),
    )
```